### PR TITLE
Move duplicate rand functionality to test rand

### PR DESCRIPTION
**Pull request by ericholguin**
This PR removes duplicate rand functions and has test files use the rand functions declared in `traffic_ops_golang/test/rand.go` removing the need to import `"math/rand"` in multiple files.

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Control Cache Config Unit Tests
- Traffic Monitor Unit Tests
- Traffic Ops Unit Tests
- Grove Unit Tests

## What is the best way to verify this PR?

Run the unit tests for the components above

## PR submission checklist
- [x] This PR has tests 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/test/rand.go [hunk 1]
```diff
@@ -36,3 +38,68 @@ func RandStrArray() []string {
...
+		uint8(rand.Int31n(256)),
+		uint8(rand.Int31n(256)),
+		uint8(rand.Int31n(256)),
+		uint8(rand.Int31n(256)),
+		uint8(rand.Int31n(256)),
+		uint8(rand.Int31n(256)),
+		uint8(rand.Int31n(256)),
+	}).String()
+	return &ip
+}
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/test/rand.go:105]: GoDocs?

**Review by ocket8888 — COMMENTED** [review]
Possible improvement: occurrences of `\*test\.Rand` are just a bit more common than those of `[^*]test\.Rand` - should these maybe return values instead of pointers?

**Comment by ericholguin** [comment]
> Possible improvement: occurrences of `\*test\.Rand` are just a bit more common than those of `[^*]test\.Rand` - should these maybe return values instead of pointers?

We would have to declare variables wherever a pointer is needed because you can't get the address of a function value returned in go. So instead of just `test.RandStr()` it'd be: 

```go
randStr := test.RandStr()
...
Fqdn:            &randStr,
...
```

And we'd either be pointing to the same random string variable where needed or create multiple random string variables and point to those. Is this what you mean or do you know of another way?

**Comment by ocket8888** [comment]
You could make use of [the `github.com/apache/trafficcontrol/lib/go-util` package](https://pkg.go.dev/github.com/apache/trafficcontrol/lib/go-util)'s functions `IntPtr`, `StrPtr`, etc. So it'd be
```go
...
Fqdn:   util.StrPtr(test.RandString())
...
```
no forward declaration required

File: traffic_ops/traffic_ops_golang/crconfig/config_test.go [hunk 2]
```diff
@@ -26,13 +26,14 @@ import (
 	"testing"
 	"time"
 
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/test"
 	"gopkg.in/DATA-DOG/go-sqlmock.v1"
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/crconfig/config_test.go:30]: By convention we generally leave a blank line between project-internal imports and external packages

File: traffic_ops/traffic_ops_golang/test/rand.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/test/rand.go]: I don't know how this got in here again. It's clearly already been in a review, and in fact already resolved.

**Pull request merged**